contents/javascript/staticTypingWithTypescript.md

```diff
@@ -0,0 +1,54 @@
+<frontmatter>
+  title: Static Typing with TypeScript
+  header: pagetop.md
+  footer: footer.md
+  head: head.md
+  siteNav: mainNav.md
+  pageNav: 3
+</frontmatter>
+
+<div class="website-content">
+
+{{ booktitle | safe }}
+
+# Static Typing with TypeScript
+
+**Author(s): [Lu Yang Kenneth](https://github.com/luyangkenneth)**
+
+Reviewer(s): WIP FILL THIS OUT LATER
+
+
+## What is Static Typing?
+
+**Statically typed languages** like Java, Go, and C++ are able to catch type-related errors at compile time, while in **dynamically typed languages** like JavaScript, such errors are not as easily discoverable because the types of variables are only known at runtime.
+
+To help identify these problems without having to run any code, many developers have adopted technologies like Flow and TypeScript, which enable static type checking for codebases written in JavaScript.
+
+Equivalent type checking tools exist for other dynamically typed languages like Python and Ruby as well, which suggests that this issue is not specific to the JavaScript community. The decision to move towards a statically typed codebase is usually motivated by the following benefits:
```

Comment:
This discussion involving other languages feel a little out of scope for the article. I feel that the motivations for static typing can stand alone without introducing static typing in other languages.

Reply:
I agree with your second point. There were a few moments where I was contemplating whether to include something because it's not strictly required but might provide useful context anyway - since these book chapters are for exposure purposes. What dyou think?

Reply:
(see below comments) I think this will feel less out of place after the rework - stay tuned!